### PR TITLE
Events and Profile updates

### LINKED ISSUE #935: Implement new mockup for user profiles
# Todos:

## Frontend
- [x] Add motivation: 
    - [x] Add edit button
    - [x] Add edit button also when user has no motivation text
- [x] Implement profile pic: 
- [x] Edit link: https://community.serlo.org/account/profile
- [x] Add link to chat
- [x] Fix link for not logged in user
- [x] Check whether user has a rocket chat link 
- [x] Add "logined since"
- [x] Add header
- [x] Link Badges to: https://de.serlo.org/community/202923/rollen-der-serlo-community
- [x] Link Roles to: https://de.serlo.org/community/202923/rollen-der-serlo-community

- [x] Add activity diagram
- [x] Add roles
- [x] Add events
- [x] Add last login
- [x] remove comments
- [x] Suggestion from Simon for better profile structuring

## Frontend: Refactoring

- [x] Refactoring CSS
- [x] Fetch motivation + imageUrl + chatUrl from API
- [x] Add semantic HTML elements for header

## Api

- [x] Add `User { imageUrl }`
- [x] Add `User { chatUrl }`
- [x] Add `User { roles }`
- [x] Add `User { events }`
- [x] add `User { motivation }`
- [x] Add statistic data for a user

## Database layer

- [x] Query for user roles
- [x] Query for events
- [x] Query for user activity 

## Infrastructure

- [x] Add changes in module-api
- [x] Add changes in env-staging
- [ ] Add changes in env-production

## Planning needed / Other
- [x] How shall we transition from the old description?
- [x] Analyze: Who has an image in its description / in the community chat? How to retrieve it?
- [x] How shall we retrieve the statistical data?
- [x] Check: Is the element "motivation" well designed?
- [x] How shall roles be displayed?

# Follow Ups:

* Upload of profile picture in Serlo.
* Add diagram for subjects: 
2021-03-08-183645_812x391_scrot
* Redirect for Rocket.Chat for not logged in user possible?

### LINKED ISSUE #1188: Events link to encoded threadId
Events relating to threads link to their encoded id which does not seem to be recognized by cf-worker so we either need to change/remove the link or investigate the cf-worker.

e.g. `dDE1NDY4`

### PR DESCRIPTION
**Pull request by elbotho**
### todo
- [x] cleanup/refactor
- [ ] test
- [ ] chase some api bugs
- [ ] maybe finish the "invite to chat" feature-mvp (depends on https://github.com/serlo/serlo.org-database-layer/issues/126)

### events
- `/event/history/{id}` basic functionality done for entities and users
- example: https://frontend-git-event-history-serlo.vercel.app/event/history/1565
- 5 events on profile pages
- ⚠️  currently not all events are provided by api

### profiles
- hide comments
- add motivation edit button
- use motivation, chatUrl and imageUrl from API
- add activity graphics
- link badges and roles to explanation page

### CODE REVIEW

File: src/fetcher/create-taxonomy.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export function buildTaxonomyData(uuid: TaxonomyTerm): TaxonomyData {
     description: uuid.description ? convertState(uuid.description) : undefined,
     title: uuid.name,
     id: uuid.id,
+    alias: uuid.alias ?? undefined,
```
Comment (kulla) [inline src/fetcher/create-taxonomy.ts:26]: This is for converting `null` into `undefined`. For which uuid we have `null` as `alias` currently?

File: src/fetcher/user/request.ts [hunk 2]
```diff
@@ -20,21 +20,14 @@ export async function requestUser(
...
         ...uuid,
-        // stub: request this from API
-        motivation:
-          uuid.username === 'Kulla'
-            ? 'Mein Ziel bei Serlo: Jedes Kind / jeder Jugendliche soll unabhängig der finanziellen Möglichkeiten seiner Familie optimal auf seinem Bildungsweg unterstützt werden.'
-            : undefined,
-        chatUrl: `https://community.serlo.org/direct/${chatName}`,
-        imageUrl: `https://community.serlo.org/avatar/${chatName}`,
+        motivation: uuid.motivation ?? undefined,
+        chatUrl: uuid.chatUrl ?? undefined,
```
Comment (kulla) [inline src/fetcher/user/request.ts:29]: Where is the conversion from `null` to `undefined` necessary?

File: src/fetcher/create-taxonomy.ts (hunk 1 above)
Comment (elbotho) [reply src/fetcher/create-taxonomy.ts:26]: Don't know but the API type `alias?: Maybe<Scalars['String']>;` allows it so as the consumer we have to work with this :)

File: src/fetcher/user/request.ts (hunk 2 above)
Comment (elbotho) [reply src/fetcher/user/request.ts:29]: Not necessary. It's just very common in the frontend to just have optional types like `chatUrl?: string`. 
But actually in this case we sometimes added the `| null` in `data-types.ts`

File: src/components/content/link.tsx [hunk 3]
```diff
@@ -53,7 +52,14 @@ export function isLegacyLink(_href: string) {
...
 
-export function Link({
+export function Link(props: LinkProps) {
+  return UnstyledLink({
+    ...props,
+    className: clsx(props.className, 'serlo-link'),
+  })
+}
+
+export function UnstyledLink({
```
Comment (elbotho) [inline src/components/content/link.tsx:62]: exporting multiple components from one file is forbidden by inyono law, sorry ;)

File: src/components/content/spoiler.tsx [hunk 4]
```diff
@@ -31,3 +31,63 @@ export function Spoiler({ body, title, path }: SpoilerProps) {
     </div>
   )
 }
+
+function SpoilerTitle({
```
Comment (elbotho) [inline src/components/content/spoiler.tsx:35]: Sorry, same here. 
If you want to introduce exporting multiple components from one file we would have to discuss this first.

File: src/components/user/event.tsx [hunk 5]
```diff
@@ -262,41 +227,76 @@ export function Notification({
...
+
+  function renderMuteButton() {
+    return (
+      <Tippy
+        duration={[300, 250]}
+        animation="fade"
+        placement="bottom"
+        content={<Tooltip>{loggedInStrings?.hide}</Tooltip>}
+      >
+        <StyledButton href={`/unsubscribe/${event.objectId.toString()}`}>
```
Comment (elbotho) [inline src/components/user/event.tsx:293]: TODO: This needs to use the new mutation.

**Comment by elbotho** [comment]
👻  Thank you @kulla for the improvements. Left some comments. 👻

File: src/components/content/spoiler.tsx (hunk 4 above)
Comment (kulla) [reply src/components/content/spoiler.tsx:35]: It's not exported though...

**Review by kulla — APPROVED** [review]
After two days of rectoring I decided to do the refactoring in the following days in https://github.com/serlo/frontend/pull/1191 This PR I push to staging so that it can be tested there.

**Pull request merged**